### PR TITLE
Fix function pointers and enums printing in complex types

### PR DESCRIPTION
**Pull request by ret2libc**

### CODE REVIEW

**Comment by ret2libc** [comment]
@XVilka i think the test will fail. I was trying to create the "pretty_print" function and i noticed that the code for `tc` command handles array, pointers and callable in a different way, while the code of `rz_type_db_base_type_as_string` does not.... Are you sure we really need a separate function for pretty print? It seems to me that `rz_type_db_base_type_as_string` is just called in a few places.

**Comment by XVilka** [comment]
Expected:
```
struct bla { int a; wchar_t (*func)(int a, const char *b); }
```
Got
```
struct bla { int a; wchar_t (*func)(int a, const char * b); func;  }
```

File: librz/type/function.c [hunk 1]
```diff
@@ -428,7 +485,13 @@ RZ_API RZ_OWN char *rz_type_callable_ptr_as_string(const RzTypeDB *typedb, RZ_NO
...
 RZ_API RZ_OWN char *rz_type_callable_as_string(const RzTypeDB *typedb, RZ_NONNULL const RzCallable *callable) {
 	rz_return_val_if_fail(typedb && callable, NULL);
-	return callable_as_string(typedb, callable, false);
+	RzStrBuf *buf = rz_strbuf_new("");
+	if (!callable_as_string(buf, typedb, callable, 0)) {
+		rz_strbuf_free(buf);
+		return NULL;
+	}
+	char *result = rz_strbuf_drain(buf);
+	return result;
```
Comment (ret2libc) [inline librz/type/function.c]:
```suggestion
	return rz_strbuf_drain(buf);
```

File: librz/type/function.c [hunk 2]
```diff
@@ -413,11 +463,18 @@ RZ_API RZ_OWN char *rz_type_callable_ptr_as_string(const RzTypeDB *typedb, RZ_NO
...
+	if (!callable) {
+		return NULL;
+	}
+	RzStrBuf *buf = rz_strbuf_new("");
+	if (!callable_as_string(buf, typedb, callable, ptr_depth)) {
+		rz_strbuf_free(buf);
+		return NULL;
 	}
+	char *result = rz_strbuf_drain(buf);
+	return result;
```
Comment (ret2libc) [inline librz/type/function.c]:
```suggestion
	return rz_strbuf_drain(buf);
```

File: librz/type/function.c [hunk 3]
```diff
@@ -374,17 +388,55 @@ RZ_API bool rz_type_is_callable_ptr(RZ_NONNULL const RzType *type) {
 	return false;
 }
 
-static RZ_OWN char *callable_as_string(const RzTypeDB *typedb, RZ_NONNULL const RzCallable *callable, bool is_ptr) {
-	rz_return_val_if_fail(typedb && callable, NULL);
+/**
+ * \brief Checks if the RzType is the nested pointer to the RzCallable
```
Comment (ret2libc) [inline librz/type/function.c:389]: what does it mean? can you make an example of what kind of "thing" would return true and what would return false?

File: librz/type/function.c [hunk 4]
```diff
@@ -359,10 +359,24 @@ RZ_API bool rz_type_func_ret_set(RzTypeDB *typedb, const char *name, RZ_OWN RZ_N
...
  * \param typedb Types Database instance
- * \param type RzCallable
+ * \param type RzType
+ */
+RZ_API bool rz_type_is_callable(RZ_NONNULL const RzType *type) {
+	rz_return_val_if_fail(type, false);
+	if (type->kind == RZ_TYPE_KIND_CALLABLE) {
+		return true;
+	}
+	return false;
```
Comment (ret2libc) [inline librz/type/function.c]:
```suggestion
	return type->kind == RZ_TYPE_KIND_CALLABLE;
```

**Review by XVilka — APPROVED** [review]

**Pull request merged**